myTrip/photo/models.py

```diff
@@ -28,41 +38,45 @@ def get_by_id(photo_id):
         Args:
             photo_id (int): photo id.
         Returns:
-            Object<Photo>: Object of Photo.
+            Object<Photo>: Object of Photo or None if got exception.
         """
         try:
             return Photo.objects.get(id=photo_id)
         except ObjectDoesNotExist:
             return None
 
     @staticmethod
-    def get_by_trip_id(trip_id):
+    def filter(trip_id, checkpoint_id):
         """
-        Get photo with given trip id
+        Get photo with given trip and checkpoint id
         Args:
-            trip_id (int): trip id.
+            trip_id (int): trip id
+            checkpoint_id (int): checkpoint id.
         Returns:
-            QuerySet<Photo>: QuerySet of Photo.
+            QuerySet<Photos>: QuerySet of Photos.
         """
-        photos = Photo.objects.filter(trip_id=trip_id)
-        return photos
+        try:
+            return Photo.objects.filter(trip_id=trip_id, checkpoint_id=checkpoint_id)
+        except ObjectDoesNotExist:
+            return None
 
     @staticmethod
-    def create(src, user_id, trip_id=None, checkpoint_id=None, description=None):
+    def create(src, user, description, trip=None, checkpoint=None):
         """ Creating photo model, and returns created object"""
         photo = Photo()
         photo.src = src
-        photo.user_id = user_id
-        photo.trip_id = trip_id
-        photo.checkpoint_id = checkpoint_id
+        photo.trip = trip
+        photo.checkpoint = checkpoint
+        photo.user = user
         photo.description = description
         photo.save()
         return photo
 
-    def update(self, description=None):
-        """Updating photo model."""
+    def update(self, description):
+        """Updating photo description."""
         if description:
             self.description = description
+            self.updated_at = datetime.now()
```

Comment:
No need to update this field!
`updated_at = models.DateTimeField(auto_now=True, null=True, editable=True)`
https://docs.djangoproject.com/en/1.11/ref/models/fields/#django.db.models.DateField.auto_now